### PR TITLE
rename platformName annotation to jvmName

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: libraries/stdlib/src/kotlin/jvm/JvmPlatformAnnotations.kt [hunk 1]
```diff
@@ -35,6 +35,16 @@ target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY, AnnotationTarget.FI
 public annotation(retention = AnnotationRetention.RUNTIME, mustBeDocumented = true) class jvmStatic
 
 /**
+ * Specifies the name for the target platform element (Java method, JavaScript function)
```
Comment (udalov) [inline libraries/stdlib/src/kotlin/jvm/JvmPlatformAnnotations.kt]: Documentation should probably not mention JavaScript anymore

File: compiler/frontend.java/src/org/jetbrains/kotlin/resolve/jvm/diagnostics/DefaultErrorMessagesJvm.java [hunk 2]
```diff
@@ -50,6 +52,8 @@ public String render(@NotNull ConflictingJvmDeclarationsData data) {
         MAP.put(ErrorsJvm.OVERLOADS_WITHOUT_DEFAULT_ARGUMENTS, "''jvmOverloads'' annotation has no effect for methods without default arguments");
         MAP.put(ErrorsJvm.OVERLOADS_ABSTRACT, "''jvmOverloads'' annotation cannot be used on abstract methods");
         MAP.put(ErrorsJvm.OVERLOADS_PRIVATE, "''jvmOverloads'' annotation has no effect on private and local declarations");
+        MAP.put(ErrorsJvm.INAPPLICABLE_JVM_NAME, "''jvmName ''annotation is not applicable to this declaration");
```
Comment (udalov) [inline compiler/frontend.java/src/org/jetbrains/kotlin/resolve/jvm/diagnostics/DefaultErrorMessagesJvm.java]: Mistyped whitespace

File: libraries/stdlib/src/kotlin/jvm/JvmPlatformAnnotations.kt (hunk 1 above)
Comment (yole) [inline libraries/stdlib/src/kotlin/jvm/JvmPlatformAnnotations.kt]: Fixed

File: compiler/frontend.java/src/org/jetbrains/kotlin/resolve/jvm/diagnostics/DefaultErrorMessagesJvm.java (hunk 2 above)
Comment (yole) [inline compiler/frontend.java/src/org/jetbrains/kotlin/resolve/jvm/diagnostics/DefaultErrorMessagesJvm.java]: Fixed

**Comment by yole** [comment]
Merged manually.

**Pull request closed without merging**